### ISSUE TITLE
Enh: Count applied/reverted migrations in case of error

### ISSUE THREAD

**Issue by Vovan-VE** [yiisoft/yii2#14169]
Currently MigrateController outputs this

> 2 from 5 migrations were applied

only when a migration does `return false;`. But in case of error (uncaught exception for example due to some stupid typos) there is no such output. If I want to undo migrations just applied (2 from 5), I need to count them manually in the output. It's annoying at least or sometimes may be impossible due too flood produced by each real query.

The suggestion is either to show same summary "2 from 5 migrations were applied" in case of error or output `"$index/$count"` on each processed migration.

### What steps will reproduce the problem?

1. Create N migrations.
2. One migration somewhere in the middle (7th for example) is bad due to some typos.
3. `./yii migrate/up`

### What is the expected result?

```
*** applying foo
    ...
*** applyed foo

*** applying bar
6 of 10 migrations were applied
7th one was failed with error
Exception 'Exception' with message 'message'

in ...
```

### What do you get instead?

```
*** applying foo
    ...
*** applyed foo

*** applying bar
Exception 'Exception' with message 'message'

in ...
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.11.2

**Comment by njasm** [comment]
what about wrapping all migrations Up or Down inside a transaction, and make them atomic? would that be an acceptable solution?

**Comment by Vovan-VE** [comment]
@njasm , no, at least until MySQL would run set of DDL statements under transaction.
